### PR TITLE
Reorganize search a bit and allow users 1 retry on catching blobs

### PR DESCRIPTION
**Pull request by scarletcafe**
Finishes a to-do in the feature/Energy project.
This moves the message calculation and catch processing out of the main command to shorten the main body slightly, and on a 2/3 allows users a second chance on catching a blob if they have the required Pokeballs to do it.

### CODE REVIEW

File: commands/search.js [hunk 1]
```diff
@@ -14,6 +14,44 @@ class Search extends Command {
...
+
+  async waitForCatchResponse(message, pokeBalls, escapedPrefix) {
+    const transform = s => s.toLowerCase().replace(/ /g, '');
+    const pokeBallNames = pokeBalls.map(x => transform(x.name)).concat(['']);
+
+    const re = new RegExp(`^${escapedPrefix}(catch|cancel|search)(.*)$`);
+    const filter = m => (m.author.id === message.author.id && re.test(m.content) && pokeBallNames.includes(transform(re.exec(m.content)[2])));
+
+    let response;
+    try {
```
Comment (7bitlyrus) [inline commands/search.js:41]: I'm pretty sure there is a better way to handle this then a try/catch, but I'm not sure

**Review by 7bitlyrus — APPROVED** [review]
Looks pretty good

File: commands/search.js (hunk 1 above)
Comment (scarletcafe) [reply commands/search.js:41]: The [discord.js documentation on awaitMessages](https://discord.js.org/#/docs/main/master/class/TextChannel?scrollTo=awaitMessages) shows using ` .catch` on a Promise. In async/await syntax, this [throws an exception so we have to catch it.](https://javascript.info/async-await#error-handling)

Comment (7bitlyrus) [reply commands/search.js:41]: Alright, fair

**Review by OGNova — APPROVED** [review]

**Pull request merged**